### PR TITLE
add signMessage precompile

### PR DESCRIPTION
**Pull request by dmarzzz**
## 📝 Summary

adds precompile to sign arbitrary digests using ECDSA signing key.

_Note: I added this precompile to `contracts_suave.go` since it is not ethereum specific._

## 📚 References

[SUAVE-spec PR](https://github.com/flashbots/suave-specs/pull/71)

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

**Review by ferranbt — DISMISSED** [review]

**Review by ferranbt — COMMENTED** [review]
Is there a conflict with #123?

**Comment by dmarzzz** [comment]
> Is there a conflict with #123?

No different curve, and ECDSA recovery method is native to solidity so don't need all of them for this PR as in #123

**Comment by dmarzzz** [comment]
yeah it's the same, I left a comment over there. I dont think we need to specify the curve in the precompile name

File: core/vm/contracts_suave.go [hunk 1]
```diff
@@ -115,6 +116,20 @@ func (b *suaveRuntime) fetchDataRecords(targetBlock uint64, namespace string) ([
 	return records, nil
 }
 
+func (s *suaveRuntime) signMessage(digest []byte, signingKey string) ([]byte, error) {
```
Comment (ferranbt) [inline core/vm/contracts_suave.go:119]: Why is `signingKey` not bytes as well?

Comment (dmarzzz) [reply core/vm/contracts_suave.go:119]: the `signingKey` in our other precompiles are also strings. Im not entirely sure why we set this precedent but I believe it was @Ruteri who did

Comment (dmarzzz) [reply core/vm/contracts_suave.go:119]: https://github.com/flashbots/suave-specs/blob/main/specs/rigil/precompiles.md#signethtransaction

**Review by ferranbt — APPROVED** [review]

**Pull request merged**